pkg/virt-launcher/virtwrap/live-migration-target.go

```diff
@@ -20,32 +20,107 @@
+	netvmispec "kubevirt.io/kubevirt/pkg/network/vmispec"
 	"kubevirt.io/kubevirt/pkg/util"
 	"kubevirt.io/kubevirt/pkg/util/net/ip"
 	migrationproxy "kubevirt.io/kubevirt/pkg/virt-handler/migration-proxy"
 	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/cli"
 	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/converter"
 )
 
-func (l *LibvirtDomainManager) finalizeMigrationTarget(vmi *v1.VirtualMachineInstance) error {
+func (l *LibvirtDomainManager) finalizeMigrationTarget(vmi *v1.VirtualMachineInstance, options *cmdv1.VirtualMachineOptions) error {
+	interfaceMigrationOptions := options.InterfaceMigration
+	if err := l.reconnectGuestNics(vmi, interfaceMigrationOptions); err != nil {
+		return err
+	}
+
 	if err := l.setGuestTime(vmi); err != nil {
 		return err
 	}
 
 	return nil
 }
 
+// Set interfaces link down and up to renew DHCP leases
+func (l *LibvirtDomainManager) reconnectGuestNics(vmi *v1.VirtualMachineInstance, interfaceMigrationOptions map[string]*cmdv1.InterfaceBindingMigration) error {
+	domName := api.VMINamespaceKeyFunc(vmi)
+	dom, err := l.virConn.LookupDomainByName(domName)
+	if err != nil {
+		log.Log.Object(vmi).Reason(err).Error("failed to reconnect guest interfaces")
+		return err
+	}
+	defer dom.Free()
+	xmlstr, err := dom.GetXMLDesc(0)
+	if err != nil {
+		return err
+	}
+
+	var domain api.DomainSpec
+	err = xml.Unmarshal([]byte(xmlstr), &domain)
+	if err != nil {
+		return fmt.Errorf("parsing domain XML failed, err: %v", err)
+	}
+
+	// Look up all the interfaces and reconnect them
+	for _, iface := range domain.Devices.Interfaces {
+		specIface := netvmispec.LookupInterfaceByName(vmi.Spec.Domain.Devices.Interfaces, iface.Alias.GetName())
+
+		migrationOption, _ := interfaceMigrationOptions[specIface.Name]
+		if migrationOption == nil || migrationOption.Method != string(v1.LinkRefresh) {
+			continue
+		}
+
+		err = reconnectIface(dom, iface)
+
+		if err != nil {
+			return fmt.Errorf("failed to update network %s, err: %v", iface.Alias.GetName(), err)
+		}
+	}
+
+	return nil
+}
+
+// Sets link down and up for specified interface
+func reconnectIface(dom cli.VirDomain, iface api.Interface) error {
+	ifaceBytes, err := xml.Marshal(iface)
+	if err != nil {
+		return fmt.Errorf("failed to encode (xml) interface, err: %v", err)
+	}
+	disconnectedIface := iface.DeepCopy()
+	disconnectedIface.LinkState = &api.LinkState{State: "down"}
+	disconnectedIfaceBytes, err := xml.Marshal(disconnectedIface)
+	if err != nil {
+		return fmt.Errorf("failed to encode (xml) interface, err: %v", err)
+	}
+	err = dom.UpdateDeviceFlags(string(disconnectedIfaceBytes), affectDeviceLiveAndConfigLibvirtFlags)
+	if err != nil {
+		return fmt.Errorf("failed to set link down, err: %v", err)
+	}
+
+	// the sleep is needed since setting the interface immediately back to up may cause some guest OSs to ignore the both down and up requests
+	time.Sleep(100 * time.Millisecond)
```

Comment:
Is it possible to "act on event" instead?

For example, for SR-IOV unplug, since detaching a host device is asynchronous we had to wait an event to verify the subject iface has been actually detached
https://github.com/kubevirt/kubevirt/blob/e4da573b08ca9238d1e6d23675ed5602f0919f57/pkg/virt-launcher/virtwrap/device/hostdevice/hotplug.go#L51-L71

Reply:
I though maybe libvirt raise events in regard to interfaces state, but setting an iface down may not affect the HW state (libvirt could monitor).